api/v1beta1/cluster_types.go

```diff
@@ -238,6 +242,42 @@ type MachineHealthCheckTopology struct {
 	MachineHealthCheckClass `json:",inline"`
 }
 
+// MachinePoolTopology specifies the different parameters for a pool of worker nodes in the topology.
+// This pool of nodes is managed by a MachinePool object whose lifecycle is managed by the Cluster controller.
+type MachinePoolTopology struct {
+	// Metadata is the metadata applied to the MachinePool.
+	// At runtime this metadata is merged with the corresponding metadata from the ClusterClass.
+	// +optional
+	Metadata ObjectMeta `json:"metadata,omitempty"`
+
+	// Class is the name of the MachinePoolClass used to create the pool of worker nodes.
+	// This should match one of the deployment classes defined in the ClusterClass object
+	// mentioned in the `Cluster.Spec.Class` field.
+	Class string `json:"class"`
+
+	// Name is the unique identifier for this MachinePoolTopology.
+	// The value is used with other unique identifiers to create a MachinePool's Name
+	// (e.g. cluster's name, etc). In case the name is greater than the allowed maximum length,
+	// the values are hashed together.
+	Name string `json:"name"`
+
+	// FailureDomains is the failure domains the machine pool will be created in.
+	// Must match a key in the FailureDomains map stored on the cluster object.
+	// +optional
+	FailureDomains []string `json:"failureDomains,omitempty"`
+
+	// Replicas is the number of nodes belonging to this pool.
+	// If the value is nil, the MachinePool is created without the number of Replicas (defaulting to 1)
+	// and it's assumed that an external entity (like cluster autoscaler) is responsible for the management
+	// of this value.
+	// +optional
+	Replicas *int32 `json:"replicas,omitempty"`
+
+	// Variables can be used to customize the MachinePool through patches.
```

Comment:
This makes sense! I excluded them since Richard's original implementation did not carry them over. I think they are actually necessary when calculating the desired state for machine pools

Reply:
Looks good here. Let's add the timeouts and MinReadySeconds to MachinePoolClass in the ClusterClass as well